src/dom-utils.ts

```diff
@@ -99,20 +99,50 @@ function getNextSiblingEl(el: HTMLElement, forward: boolean) {
 }
 
 /**
- * Determine if an element is focusable and has user-visible painted dimensions.
+ * Determine if an element is hidden from the user.
  */
-export function isFocusable(el: HTMLElement) {
-  return (
-    el.matches(focusableSelectors.join(',')) &&
-    !!(el.offsetWidth || el.offsetHeight || el.getClientRects().length)
+const isHidden = (el: HTMLElement) => {
+  /**
+   * Browsers hide all non-<summary> descendants of closed <details> elements
+   * from user interaction, but those non-<summary> elements may still match our
+   * focusable-selectors and may still have dimensions, so we need a special
+   * case to ignore them.
+   */
+  if (
+    el.matches('details:not([open]) *') &&
```

Comment:
Maybe `el.closest('details:not([open])')`? What do you think?

Reply:
I would prefer that we stick to `.match()`. The difference is small, but I imagine `.match()` will be slightly more performant because it only has to check the immediate node, where `.closest()` has to check ancestors of the node all the way up the tree.